packages/pages-shared/asset-server/handler.ts

```diff
@@ -516,32 +530,46 @@ export async function generateHandler<
 				response.headers.set("x-robots-tag", "noindex");
 			}
 
-			if (options.preserve) {
-				// https://fetch.spec.whatwg.org/#null-body-status
-				const preservedResponse = new Response(
-					[101, 204, 205, 304].includes(response.status)
-						? null
-						: response.clone().body,
-					response
-				);
-				preservedResponse.headers.set(
-					"cache-control",
-					CACHE_CONTROL_PRESERVATION
-				);
-				preservedResponse.headers.set("x-robots-tag", "noindex");
+			if (options.preserve && waitUntil && caches) {
+				waitUntil(
+					(async () => {
+						try {
+							const assetPreservationCacheV2 = await caches.open(
+								ASSET_PRESERVATION_CACHE_V2
+							);
 
-				if (waitUntil && caches) {
-					waitUntil(
-						caches
-							.open(ASSET_PRESERVATION_CACHE)
-							.then((assetPreservationCache) =>
-								assetPreservationCache.put(request.url, preservedResponse)
-							)
-							.catch((err) => {
-								logError(err);
-							})
-					);
-				}
+							let shouldUpdateCache = true;
```

Comment:
I'm not sure why we bother with a variable here? It is only assigned in one place and only read directly afterwards. I think we could ditch this and simply do something like:

```ts
if (assetKey !== await match?.text()) {
```

Reply:
That's fair, updated like this:
```typescript
const match = await assetPreservationCacheV2.match(request);
if (match) {
	if (assetKey !== (await match.text())) {
```